lib/csv.rb

```diff
@@ -854,6 +855,10 @@ def initialize(message, line_number)
     end
   end
 
+  # The error thrown when the parser is given mutually incompatible options.
+  class InvalidOptionsError < RuntimeError
+  end
```

Comment:
Could you use the builtin `ArgumentError` instead of creating a new error class? It's suitable for this case.

Reply:
Yes, you're quite right—I've now updated the pull request.